File: longestCommonPrefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.size() == 0) {
            return "";
        }
        
        return getLongestCommonPrefix(strs, 0, strs.size()-1);
    }
    
    string getLongestCommonPrefix(vector<string>& strs, int l, int r) {
        if (l == r) {
            return strs[l];
        }
        else {
            int mid = (l + r)/2;
            string left = getLongestCommonPrefix(strs, l, mid);
            string right = getLongestCommonPrefix(strs, mid+1, r);
            return commonPrefix(left, right);
        }
    }
    
    string commonPrefix(string left, string right) {
        string commonPrefix = "";
        for (int i = 0; i < min(left.size(), right.size()); ++i) {
            cout << left[i] << " ";
            if (left[i] == right[i]) {
                commonPrefix+=left[i];
            }
            else {
                break;
            }
        }
        return commonPrefix;
    }
};

File: LongestCommonPrefix.py
class Solution:
    """
    @param strs: A list of strings
    @return: The longest common prefix
    """
    def longestCommonPrefix(self, strs):
        if strs == []:
            return ""
        if len(strs) == 1:
            return strs[0]
            
        index = 0
        while 1:
            if self.ifithequal(strs, index):
                index += 1
                continue
            break
        
        return strs[0][:index]
                
    def ifithequal(self, strs, i):
        if i >= len(strs[0]):
            return False
        char = strs[0][i]
        for str in strs:
            if i >= len(str):
                return False
            if str[i] != char:
                return False
        return True
                
